jwst/extract_1d/soss_extract/pastasoss.py

```diff
@@ -0,0 +1,537 @@
+from functools import partial
+from typing import Tuple
+import logging
+import numpy as np
+from scipy.interpolate import interp1d
+from ...datamodels import PastasossModel
+
+log = logging.getLogger(__name__)
+
+
+def get_wavelengths(
+    refmodel: PastasossModel, x: np.ndarray, pwcpos: float, order: int
+) -> np.ndarray:
+    """Get the associated wavelength values for a given spectral order"""
+    if order == 1:
+        wavelengths = wavecal_model_order1_poly(refmodel, x, pwcpos)
+    elif order == 2:
+        wavelengths = wavecal_model_order2_poly(refmodel, x, pwcpos)
+
+    return wavelengths
+
+
+def min_max_scaler(x, x_min, x_max):
+    """
+    Apply min-max scaling to input values.
+
+    Parameters
+    ----------
+    x : float or numpy.ndarray
+        The input value(s) to be scaled.
+    x_min : float
+        The minimum value in the range to which 'x' will be scaled.
+    x_max : float
+        The maximum value in the range to which 'x' will be scaled.
+
+    Returns
+    -------
+    float or numpy.ndarray
+        The scaled value(s) in the range [0, 1].
+
+    Notes
+    -----
+    Min-max scaling is a data normalization technique that scales input values
+    'x' to the range [0, 1] based on the provided minimum and maximum values,
+    'x_min' and 'x_max'. This function is applicable to both individual values
+    and arrays of values. This function will use the min/max values from the
+    training data of the wavecal model.
+    """
+    # scaling the input x values
+    x_scaled = (x - x_min) / (x_max - x_min)
+    return x_scaled
+
+
+def wavecal_model_order1_poly(refmodel, x, pwcpos):
```

Comment:
add short docstring with parameter definitions, as it wasn't immediately obvious to me what `pwcpos` was at a glance